lib/spack/spack/build_systems/autotools.py

```diff
@@ -196,20 +196,21 @@ def _do_patch_libtool(self):
         detect the compiler (and patch_libtool is set), patch in the correct
         flags for the Arm, Clang/Flang, and Fujitsu compilers."""
 
-        libtool = os.path.join(self.build_directory, "libtool")
-        if self.patch_libtool and os.path.exists(libtool):
-            if self.spec.satisfies('%arm') or self.spec.satisfies('%clang') \
-                    or self.spec.satisfies('%fj'):
-                for line in fileinput.input(libtool, inplace=True):
-                    # Replace missing flags with those for Arm/Clang
-                    if line == 'wl=""\n':
-                        line = 'wl="-Wl,"\n'
-                    if line == 'pic_flag=""\n':
-                        line = 'pic_flag="{0}"\n'\
-                               .format(self.compiler.cc_pic_flag)
-                    if self.spec.satisfies('%fj') and 'fjhpctag.o' in line:
-                        line = re.sub(r'/\S*/fjhpctag.o', '', line)
-                    sys.stdout.write(line)
+        for libtool in find(self.build_directory, 'libtool', recursive=True):
+            if self.patch_libtool:
```

Comment:
Here we are calling a potentially expensive find to check only afterwards if it is necessary. I would invert the logic:
```suggestion
        # Exit early if we are required not to patch libtool
        if not self.patch_libtool:
            return
            
        for libtool in find(self.build_directory, 'libtool', recursive=True):
```

Reply:
I think this fix makes more sense. I fixed it.